crazyflow/sim/visualize.py

```diff
@@ -69,6 +69,61 @@ def draw_points(sim: Sim, points: NDArray, rgba: NDArray | None = None, size: fl
         )
 
 
+def change_material(
+    sim: Sim,
+    mat_name: str,
+    drone_ids: NDArray,
+    rgba: NDArray | None = None,
+    emission: NDArray | None = None,
+):
+    """Change the material of all drones matching the mask.
+
+    Args:
+        sim: The simulation.
+        mat_name: The name of the material to change.
+        drone_ids: Array of drone indices to modify, shape (n,), dtype=int.
+        rgba: The RGBA color to set, should be of shape (n, 4) or (4,) to be auto-broadcasted.
+        emission: The emission value of material, should be of shape (n,) or scalar.
+    """
+    if drone_ids.ndim != 1:
+        raise ValueError(f"drone_ids must be 1D array, got shape {drone_ids.shape}")
+    if np.any(drone_ids < 0) or np.any(drone_ids >= sim.n_drones):
+        raise ValueError(f"drone_ids must be in range [0, {sim.n_drones - 1}], got {drone_ids}")
+
+    if rgba is not None:
+        rgba = np.asarray(rgba, dtype=float)
+        try:
+            # this returns itself if rgba is already the right shape
+            rgba = np.broadcast_to(rgba, (len(drone_ids), 4)).copy()
+        except Exception:
+            raise ValueError(f"rgba must be shape (4,) or ({len(drone_ids)}, 4), got {rgba.shape}")
+        rgba = np.clip(rgba, 0.0, 1.0)
+
+    if emission is not None:
+        emission = np.asarray(emission, dtype=float)
+        try:
+            emission = np.broadcast_to(emission, (len(drone_ids),)).copy()
+        except Exception:
+            raise ValueError(
+                f"emission must be scalar or shape ({len(drone_ids)},), got {emission.shape}"
+            )
+        emission = np.maximum(emission, 0.0)
+
+    mj_model = sim.mj_model
+
+    for i, id in enumerate(drone_ids):
```

Comment:
Do not use id, it's a built-in function that you are shadowing (https://docs.python.org/3/library/functions.html#id).